tcpMaxConn.go

```diff
@@ -3,41 +3,68 @@ package main
 import (
 	"flag"
 	"fmt"
+	"github.com/dachad/check-max-tcp-connections/tcpclient"
+	"io"
+	"io/ioutil"
+	"os"
 	"runtime"
 	"strconv"
 	"sync"
 	"time"
-	"github.com/dachad/check-max-tcp-connections/tcpclient"
 )
 
-func runTcpConnectionsInParallel(numberConnections int, delay int, host string, port int) {
-	runtime.GOMAXPROCS(numberConnections)
+var debugOut io.Writer = ioutil.Discard
+
+func runTcpConnectionsInParallel(numberConnections int, delay int, host string, port int, rinterval int) {
+	runtime.GOMAXPROCS(numberConnections+2)
+	connStatusCh := make(chan tcpclient.Connection, numberConnections*3)
+	connStatusTracker := make([]tcpclient.Connection, numberConnections)
+
```

Comment:
I don't see any point where the `ConnectionNotInitiated` is used...

Reply:
Uninitialised ints in go get a 0 value. So, unless we pick this connection to start dialing, it will remain in this status. Nice trick, yeah?! ;)

Reply:
yup! something new to learn xD